timesketch/lib/graphs/chrome_downloads.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Graph plugin for Chrome downloads."""
+
+from timesketch.lib.graphs.interface import BaseGraphPlugin
+from timesketch.lib.graphs import manager
+
+
+class ChromeDownloadsGraph(BaseGraphPlugin):
+    """Graph plugin for Chrome downloads and executions.
+
+    This plugin will try to match any downloaded files with Windows prefetch
+    executions.
+    """
+
+    NAME = 'ChromeDownloads'
+    DISPLAY_NAME = 'Chrome downloads'
+
+    def generate(self):
+        """Generate the graph.
+
+        Returns:
+            Graph object instance.
+        """
+        query = 'data_type:"chrome:history:file_downloaded"'
+        return_fields = [
+            'hostname', 'received_bytes', 'full_path', 'url'
+        ]
+
+        chrome_events = self.event_stream(
+            query_string=query, return_fields=return_fields, indices=['_all'])
+
+        for chrome_event in chrome_events:
+            computer_name = chrome_event['_source'].get('hostname', '')
+            full_path = chrome_event['_source'].get('full_path', '')
+            received_bytes = chrome_event['_source'].get('received_bytes', '')
+            url = chrome_event['_source'].get('url', '')
+
+            if '\\' in full_path:
+                separator = '\\'
+            else:
+                separator = '/'
+            filename = full_path.split(separator)[-1]
+            computer = self.graph.add_node(computer_name, {'type': 'computer'})
+
+            # If full patch is missing, set url as label.
+            if not full_path:
+                full_path = url
+                filename = url
+
+            file = self.graph.add_node(filename, {
+                'full_path': full_path,
+                'received_bytes': received_bytes,
+                'type': 'file'
+            })
+
+            edge_label = 'Downloaded {0:d} bytes'.format(received_bytes)
```

Comment:
why not use f'' strings here

Reply:
same with other strings here

Reply:
Fixed here, and all other format strings.